src/client/components/BaselayerControl.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <div
+    class="baselayer-control baselayer-thumbnail"
+    @click="switchBaseLayer"
+  >
+    <img
+      :src="thumbnail"
+      :alt="altText"
+      class="baselayer-thumbnail__image"
+    >
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    layers: {
+      type: Array,
+      required: true
+    },
+    switchHandler: {
+      type: Function,
+      required: true
+    }
+  },
+  data() {
+    return {
+      nextLayerIndex: 1
+    }
+  },
+  computed: {
+    altText() {
+      return `${this.layers[this.nextLayerIndex].title} thumbnail` || 'thumbnail'
+    },
+    thumbnail() {
+      return this.layers[this.nextLayerIndex].thumbnail || ''
+    }
+  },
+  methods: {
+    switchBaseLayer() {
```

Comment:
`switchBaseLayer` -> `cycleLayers` / `switchToNextLayer`?

Reply:
`switchBaseLayer` seems the most descriptive to me to be honest.

Reply:
I think I agree with Jasper. A switch can be a toggle (true/false) or it would go to a certain value (switchTo). In this case it can have more than two states yet the state it goes to is not a parameter? Then cycle makes more sense

Reply:
Then you're describing the implementation. Maybe `switchToNextBaseLayer` is the best option then.